### PR TITLE
Some new scheduling checks

### PR DESCRIPTION
**Pull request by mgersh**
* Classes that are scheduled, but aren't approved
* Classes that have gaps in them (will also catch classes that are longer than they're supposed to be)
* Fix to hungry teachers

The first three commits here are already in mit-prod-ajax-spike and were used during scheduling.

### CODE REVIEW

File: esp/esp/program/modules/handlers/schedulingcheckmodule.py [hunk 1]
```diff
@@ -233,6 +235,24 @@ def classes_which_cover_lunch(self):
                          l.append(s)
           return self.formatter.format_list(l)
 
+     def classes_too_long(self):
+         output = []
+         for sec in self._all_class_sections():
+             start_time = sec.start_time_prefetchable()
+             end_time = sec.end_time_prefetchable()
+             length = end_time - start_time
+             if length.total_seconds()/float(3600) - float(sec.duration) > 0.3:
```
Comment (benjaminjkraft) [inline esp/esp/program/modules/handlers/schedulingcheckmodule.py]: Is it worth also catching classes that are too short?

**Comment by benjaminjkraft** [comment]
:heavy_check_mark: Just one suggestion.

File: esp/esp/program/modules/handlers/schedulingcheckmodule.py (hunk 1 above)
Comment (mgersh) [inline esp/esp/program/modules/handlers/schedulingcheckmodule.py]: Done.

**Comment by benjaminjkraft** [comment]
:white_check_mark:

**Pull request merged**